CMakeLists.txt

```diff
@@ -2,22 +2,34 @@ cmake_minimum_required(VERSION 3.13)
 
 project(GView VERSION 1.0)
 
+# versioning
+file(READ "GViewCore/include/GView.hpp" GView_content)
+string(REGEX MATCH "#define GVIEW_VERSION \"([0-9]+.[0-9]+.[0-9]+)\"" _ ${GView_content})
+set(GVIEW_VERSION ${CMAKE_MATCH_1})
+message("${PROJECT_NAME} version: ${GVIEW_VERSION}")
+
 set(CMAKE_MODULE_PATH "${CMAKE_MODULE_PATH};${CMAKE_CURRENT_SOURCE_DIR}/cmake")
 
 set_property(GLOBAL PROPERTY USE_FOLDERS ON)
 set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 
+if(NOT CMAKE_BUILD_TYPE)
+    set(CMAKE_BUILD_TYPE "Debug" CACHE STRING 
+	    "Choose the type of build, options are: Debug Release RelWithDebInfo MinSizeRel." 
+	    FORCE)
+endif(NOT CMAKE_BUILD_TYPE)
+
 if (CMAKE_BUILD_TYPE STREQUAL "Debug")
      set(DEBUG_BUILD 1)
 endif()
 
 find_package(OpenSSL REQUIRED)
 
-set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/../bin)
-set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/../bin)
-set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/../bin)
-set(CMAKE_RUNTIME_OUTPUT_DIRECTORY_DEBUG ${CMAKE_BINARY_DIR}/../bin/Debug)
-set(CMAKE_RUNTIME_OUTPUT_DIRECTORY_RELEASE ${CMAKE_BINARY_DIR}/../bin/Release)
+set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/../bin/${CMAKE_BUILD_TYPE}")
```

Comment:
daca faci un folder in bin cu fiecare built type, nu mai bine faci direct un folder bin in locul unde e buildat codul?

Reply:
fiecare chestie merge separata n folderul ei
in caz ca schimbi built type

Reply:
(putem) pune un issue pentru unificarea `Build` si `bin` folders